Д\З Семинар 3

Mirremar opened the pull request:
Добавлено:
1) Класс Service для работы с созданием членов семьи и сортировкой 
2) Итератор HumanIterator для возможности прохода по коллекциям людей 
3) Компаратор HumanAgeComparator для настройки сравнения людей для сортировки по возрасту 
4) Методы сортировки по имени и возрасту

Mirremar committed:
ad

Mirremar committed:
homework done

Mirremar committed:
added comments

Mirremar committed:
added saveload tofrom file

Mirremar committed:
homework2

Mirremar committed:
Delete Homework_s1/demo directory

Mirremar committed:
Delete Homework_Attestation directory

Mirremar committed:
Delete mergesort directory

Mirremar committed:
Delete L1task2.java

Mirremar committed:
Delete L1task2testfile1.txt

Mirremar committed:
Delete L1task2testfile2.txt

Mirremar committed:
Delete L1task2testfile3.txt

Mirremar committed:
Delete Lesson1.java

Mirremar committed:
Добавлено:
1) Класс Service для работы с созданием членов семьи и сортировкой
2) Итератор HumanIterator для возможности прохода по коллекциям людей
3) Компаратор HumanAgeComparator для настройки сравнения людей для сортировки по возрасту
4) Методы сортировки по имени и возрасту

Liberate520 commented on FamilyTree.java:
```diff
@@ -0,0 +1,178 @@
+            catch (NullPointerException e) {
+
+            }
+            try {
+                Human somefather = searchHumanByName(h.getFather().getName());
+                somefather.getChildren().add(h);
+
+            }
+            catch (NullPointerException e) {
+
+            }
```
Плохой стиль. блок try-catch тут не обязателен. Правильно использовать if и проверка на null перед выполнением действия

Liberate520 replied:
тем более что блок catch пустой, это прям совсем плохо

Liberate520 commented on FamilyTree.java:
```diff
@@ -0,0 +1,178 @@
+        return results;
+    }
+
+    private int DatesPosition(Human h, LocalDate ps, LocalDate pe) { //Вспомогательный метод,определяет,
```
название методод с маленькой буквы

Liberate520 replied:
В LocalDate есть методы before и after, не надо изобретать велосипед

Liberate520 commented on FamilyTree.java:
```diff
@@ -0,0 +1,178 @@
+    public void redactRelative(Human h, String newName, String newDob, String newDod, String newGender, String newFather,
+                               String newMother) {
+
+        if (!(newName.equals(">"))) {
```
Ссылочные типы данных могут иметь значение null. Нет необходимости придумывать другие символы для указания того, что объекта нет

Mirremar committed:
1) Исправлены замечания
2) Добавлен интерфейс Entity для обобщения (для работы не только с сущностями типа Human)
3) Добавлен компаратор EntityNameComparator для сравнения объектов по именам
4) Добавлен класс ToUserCommunication. Туда из Main перенесено взаимодействие с пользователем
5) Исправлены замечания

Mirremar committed:
1) Исправлены замечания
2) Добавлен интерфейс Entity для обобщения (для работы не только с сущностями типа Human)
3) Добавлен компаратор EntityNameComparator для сравнения объектов по именам
4) Добавлен класс ToUserCommunication. Туда из Main перенесено взаимодействие с пользователем
5) Исправлены замечания
6) Создана пакетная структура проекта
